src/services/exchanges-service/exchanges-service.ts

```diff
@@ -15,8 +15,11 @@ export class ExchangesService implements IExchangesService {
     constructor(private http: HttpClient) {}
 
     getAvailableExchanges(): Promise<{availableExchanges: IAvailableExchange[]}> {
+        
         return this.http.get('assets/json/availableExchanges.json').toPromise().then(data => {
             return <any>data;
+        }).catch(data => {
```

Comment:
Why do we need a catch? You could ignore the catch and only handle the success case. Any errors will then be passed to the next Promise handler in the chain. Also the current implementation hides the error message. Error messages should not be hidden because they provide valuable information to the caller function.